### PR TITLE
CVS-82872 static shapes support without UT

### PR DESCRIPTION
**Pull request by ficol**

### CODE REVIEW

File: src/kfs_grpc_inference_service.cpp [hunk 1]
```diff
@@ -14,133 +14,221 @@
...
+    raw_contents->assign((char*)data, bytesize);
+    // raw_contents->assign((char*) outputString.c_str(), bytesize);
+    response->set_id(request->id());
+
+    return ::grpc::Status(::grpc::StatusCode::OK, "");
 }
+
+::grpc::Status KFSInferenceServiceImpl::buildResponse(
+    std::shared_ptr<ModelInstance> instance,
+    ::inference::ModelMetadataResponse* response) {
```
Comment (mkuczyns11) [inline src/kfs_grpc_inference_service.cpp:167]: Return ovms Status instead of grpc and simply map it in ModelMetadata?

Comment (mkuczyns11) [reply src/kfs_grpc_inference_service.cpp:167]: This would make testing easier.

File: src/kfs_grpc_inference_service.cpp [hunk 2]
```diff
@@ -14,133 +14,221 @@
...
+    // 0 meaning immediately return unload guard if possible, otherwise do not wait for available state
+    auto status = instance->waitForLoaded(0, unloadGuard);
+    if (!status.ok()) {
+        return ::grpc::Status(::grpc::StatusCode::UNAVAILABLE, "Failed to load model");
+    }
+
+    response->Clear();
+    response->set_name(instance->getName());
+    response->add_versions(std::to_string(instance->getVersion()));
+    response->set_platform("OpenVINO");  // here need to add DL/ML framework/backend
```
Comment (mkuczyns11) [inline src/kfs_grpc_inference_service.cpp:180]: Remove comment if resolved

File: src/kfs_grpc_inference_service.cpp [hunk 3]
```diff
@@ -14,133 +14,221 @@
...
+
+    // 0 meaning immediately return unload guard if possible, otherwise do not wait for available state
+    auto status = pipelineDefinition.waitForLoaded(unloadGuard, 0);
+    if (!status.ok()) {
+        return ::grpc::Status(::grpc::StatusCode::UNAVAILABLE, "Failed to load pipeline");
+    }
+
+    response->Clear();
+    response->set_name(pipelineDefinition.getName());
+    response->set_platform("OpenVINO");  // here need to add DL/ML framework/backend
```
Comment (mkuczyns11) [inline src/kfs_grpc_inference_service.cpp:207]: Same as above

Comment (mkuczyns11) [inline src/kfs_grpc_inference_service.cpp:207]: Version should be defined and its value should be '1'.

File: src/kfs_grpc_inference_service.hpp [hunk 4]
```diff
@@ -24,17 +24,26 @@
...
+    ::grpc::Status ModelReady(::grpc::ServerContext* context, const ::inference::ModelReadyRequest* request, ::inference::ModelReadyResponse* response) override;
+    ::grpc::Status ServerMetadata(::grpc::ServerContext* context, const ::inference::ServerMetadataRequest* request, ::inference::ServerMetadataResponse* response) override;
+    ::grpc::Status ModelMetadata(::grpc::ServerContext* context, const ::inference::ModelMetadataRequest* request, ::inference::ModelMetadataResponse* response) override;
+    ::grpc::Status ModelInfer(::grpc::ServerContext* context, const ::inference::ModelInferRequest* request, ::inference::ModelInferResponse* response) override;
+    ::grpc::Status buildResponse(std::shared_ptr<ModelInstance> instance, ::inference::ModelMetadataResponse* response);
+    ::grpc::Status buildResponse(PipelineDefinition& pipelineDefinition, ::inference::ModelMetadataResponse* response);
+    void convert(const std::pair<std::string, std::shared_ptr<TensorInfo>>& from, ::inference::ModelMetadataResponse::TensorMetadata* to);
```
Comment (mkuczyns11) [inline src/kfs_grpc_inference_service.hpp:46]: For testing purposes at least 'buildResponse' functions should be public.

**Review by mkuczyns11 — CHANGES_REQUESTED** [review]

**Review by mkuczyns11 — APPROVED** [review]

**Review by atobiszei — APPROVED** [review]

**Pull request merged**